DrawerKit/DrawerKit/Internal API/AnimationSupport.swift

```diff
@@ -0,0 +1,78 @@
+import UIKit
+
+struct AnimationSupport {
+    static func actualTransitionDuration(from startingPositionY: CGFloat,
+                                         to endingPositionY: CGFloat,
+                                         containerViewHeight: CGFloat,
+                                         configuration: DrawerConfiguration) -> TimeInterval {
+        var duration = configuration.totalDurationInSeconds
+        if configuration.durationIsProportionalToDistanceTraveled {
+            let fractionToGo = abs(endingPositionY - startingPositionY) / containerViewHeight
+            duration *= TimeInterval(fractionToGo)
+        }
+        return duration
+    }
+
+    static func makeGeometry(containerBounds: CGRect,
+                             startingFrame: CGRect,
+                             endingFrame: CGRect,
+                             presentingVC: UIViewController,
+                             presentedVC: UIViewController) -> DrawerGeometry {
+        let navigationBarHeight =
+            presentingVC.navigationController?.navigationBar.bounds.size.height ?? 0
+
+        let heightOfPartiallyExpandedDrawer =
+            (presentedVC as? DrawerPresentable)?.heightOfPartiallyExpandedDrawer ?? 0
+
+        return DrawerGeometry(containerBounds: containerBounds,
+                              startingFrame: startingFrame,
+                              endingFrame: endingFrame,
+                              navigationBarHeight: navigationBarHeight,
+                              heightOfPartiallyExpandedDrawer: heightOfPartiallyExpandedDrawer)
+    }
+
+    static func makeInfo(startDrawerState: DrawerState,
+                         targetDrawerState: DrawerState,
+                         _ configuration: DrawerConfiguration,
+                         _ geometry: DrawerGeometry,
+                         _ actualDurationInSeconds: TimeInterval,
+                         _ isPresenting: Bool,
+                         _ endingPosition: UIViewAnimatingPosition? = nil) -> DrawerAnimationInfo {
+        // XXX - what if endingPosition == .current ?
```

Comment:
What does XXX mean? Is this a TODO?

Reply:
LOL... it's a "personal" TODO, just for myself. Forgot to check for those.

Reply:
Fixed in https://github.com/Babylonpartners/DrawerKit/pull/30/commits/b8170dd261ce91215cb043b0593603e9557120f0.